sympy/tensor/array/dense_ndim_array.py

```diff
@@ -148,11 +155,12 @@ def __new__(cls, iterable, shape=None, **kwargs):
     @classmethod
     def _new(cls, iterable, shape, **kwargs):
         from sympy.utilities.iterables import flatten
+        from sympy.tensor.array.arrayop import Flatten
 
         shape, flat_list = cls._handle_ndarray_creation_inputs(iterable, shape, **kwargs)
         shape = Tuple(*map(_sympify, shape))
         cls._check_special_bounds(flat_list, shape)
-        flat_list = flatten(flat_list)
+        flat_list = flatten(Flatten(flat_list))
```

Comment:
is this really needed? isn't `flatten` already supposed to flatten the list?

Reply:
No, it can be removed. I didn't think about it. Thank you!!
Since I have modified the `flatten()` function in the last commit, it can now flatten the result of `list(Array())`. 
What we had before is that it would run into the case of having 'args' and transform an `Array` to `Arrays.args`, which will add the `shape` into the flatten list.
```python
 if hasattr(el, 'args'):
    el = el.args
```